scripts/build-dist.js

```
@@ -0,0 +1,92 @@
+        entry: p.resolve('src/react-intl.js'),
+        plugins: [
+            babel({
+                presets: [
+                    'es2015-rollup',
+                    'react',
+                ],
+            }),
+            npm({
+                jsnext: true,
+                skip: [
+                    'react',
+                ],
+            }),
+            commonjs({
+                sourceMap: true,
+            }),
+            replace({
+                'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
+            }),
+        ],
+    });
+}
+
+function writeBundle(bundle, {minify = false}) {
+    const filename = `react-intl${minify ? '.min.js' : '.js'}`;
+    const dest = p.resolve(`dist/${filename}`);
+
+    let result = bundle.generate({
+        format: 'umd',
+        moduleName: 'ReactIntl',
+        banner: copyright,
+        sourceMap: true,
+        sourceMapFile: dest,
+        globals: {
+            react: 'React',
+        },
+    });
+
+    if (minify) {
+        result = UglifyJS.minify(result.code, {
+            fromString: true,
+            inSourceMap: result.map,
+            outSourceMap: `${filename}.map`,
+            warnings: true,
+        });
+
+        result.map = JSON.parse(result.map);
+    } else {
+        result.code += `\n//# sourceMappingURL=${filename}.map`;
+    }
+
+    let {code, map} = result;
+
+    // Tweak source paths.
+    map.sources = map.sources.map((path) => p.relative('..', path));
+    map.sourceRoot = 'react-intl:///';
+
+    fs.writeFile(dest, code);
+    fs.writeFile(`${dest}.map`, JSON.stringify(map));
+
+    console.log(`Writing: ${p.relative('.', dest)}`);
+    console.log(`Writing: ${p.relative('.', `${dest}.map`)}`);
+}
+
+// -----------------------------------------------------------------------------
+
+createBundle()
+    .then((bundle) => {
+        writeBundle(bundle, {minify: process.env.NODE_ENV === 'production'});
+    })
+    .catch((err) => console.error(err));
```

Well, IMHO you may want to consider at least re-throw after you log it so that if this script gets refactored down the road to be something that returns that promise that the error propagates out since what you have hear actually does swallow the err.

Also, I wanted to make sure to toss some resources in a comment just in case somebody finds this PR and thinks (as a lot of people do) that Promises swallow errors :)